ports/atmel-samd/common-hal/pulseio/PulseIn.c

```diff
@@ -121,8 +122,14 @@ void pulsein_interrupt_handler(uint8_t channel) {
             self->start++;
         }
     }
+    if (!supervisor_background_tasks_ok() ) {
+        common_hal_mcu_enable_interrupts();
+        mp_raise_RuntimeError(translate("Input taking too long"));
```

Comment:
Please don't raise an exception in an interrupt handler because it may mess up an interrupts we're nested in and also may raise the exception with an unrelated python stack trace. self->errored_too_fast was a way to signal the issue and the calls into PulseIn could see it and raise an error when called.